src/libraries/System.Formats.Tar/src/System/Formats/Tar/PosixTarEntry.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Formats.Tar
+{
+    /// <summary>
+    /// Abstract class that represents a tar entry from an archive of a format that is based on the POSIX IEEE P1003.1 standard from 1988. This includes the formats <see cref="TarFormat.Ustar"/> (represented by the <see cref="UstarTarEntry"/> class), <see cref="TarFormat.Pax"/> (represented by the <see cref="PaxTarEntry"/> class) and <see cref="TarFormat.Gnu"/> (represented by the <see cref="GnuTarEntry"/> class).
+    /// </summary>
+    /// <remarks>Formats that implement the POSIX IEEE P1003.1 standard from 1988, support the following header fields: <c>devmajor</c>, <c>devminor</c>, <c>gname</c> and <c>uname</c>.
+    /// Even though the <see cref="TarFormat.Gnu"/> format is not POSIX compatible, it implements and supports the Unix-specific fields that were defined in that POSIX standard.</remarks>
+    public abstract partial class PosixTarEntry : TarEntry
+    {
+        // Constructor used when reading an existing archive.
+        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin)
+            : base(header, readerOfOrigin)
+        {
+        }
+
+        // Constructor called when creating a new 'TarEntry*' instance that can be passed to a TarWriter.
+        internal PosixTarEntry(TarEntryType entryType, string entryName, TarFormat format)
+            : base(entryType, entryName, format)
+        {
+        }
+
+        /// <summary>
+        /// When the current entry represents a character device or a block device, the major number identifies the driver associated with the device.
+        /// </summary>
+        /// <remarks>Character and block devices are Unix-specific entry types.</remarks>
+        /// <exception cref="InvalidOperationException">The entry does not represent a block device or a character device.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Cannot set a negative value.</exception>
```

Comment:
This comment is inconsistent with the check actually being performed.  Same for DeviceMinor below.

Reply:
Actually, the upper limit is wrong.

The devmajor and devminor fields in the header have a length of 8 bytes, and are stored as octal integers in ASCII.

_Note: the spec does not indicate that they are stored in octal, but if you use the gnu `tar` tool to store a block/char device and inspect the file with a hex editor, the numbers are stored in octal. Also, the `tar` tool puts a null char in the last byte, even though this is also not documented in the spec._

So we have 7 available bytes, which means the upper bound is `7,777,777` in octal, or `2,097,151` in decimal. I'll adjust this and fix the exception message too.

Reply:
https://www.ibm.com/docs/en/zos/2.3.0?topic=scd-tar-manipulate-tar-archive-files-copy-back-up-file
```
The POSIX 1003.1 standard defines formats for tar and USTAR archives that limit the modification time to a maximum value of 77777777777 (octal). Modification times that are larger than the maximum value are not properly saved. Modification times beyond 03:14:07 UTC on January 19, 2038 but less than or equal to the maximum value, are not properly saved or restored on z/OS V2R2 and earlier systems. They also might not be properly saved or restored on non-z/OS systems. However, on systems later than z/OS V2R2, such values are properly saved and restored. Due to these restrictions, the pax utility with the pax interchange format is the preferred method for processing time values beyond 03:14:07 UTC on January 19, 2038.
```